test test_project_changed_changes_str is flaky

This test just failed for me and then in the next test run without changing anything it passed again.

```
FAIL: test_project_changed_changes_str (coldfront.core.allocation.tests.test_models.AllocationModelStrTests.test_project_changed_changes_str)
When the project associated with this allocation changes the str should change
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/srv/simon/src/coldfront/coldfront/core/allocation/tests/test_models.py", line 215, in test_project_changed_changes_str
    self.assertNotEqual(original_string, new_string)
AssertionError: 'trial/according (MaryGalloway)' == 'trial/according (MaryGalloway)'
```

--- a/coldfront/core/allocation/tests/test_models.py
+++ b/coldfront/core/allocation/tests/test_models.py
@@ -322,7 +322,7 @@ class AllocationModelStrTests(TestCase):
         """When the project associated with this allocation changes the str should change"""
         original_string = str(self.allocation)
 
-        new_project = ProjectFactory()
+        new_project = ProjectFactory(pi=UserFactory(first_name="Unique Name"))
         self.allocation.project = new_project
         self.allocation.save()
 

